### ISSUE TITLE
Assertion wrongly fails if asserting with to.not.be before asserting with to.be

### ISSUE THREAD

**Issue by dannyhajj** [a85/POSTMan-Chrome-Extension#3592]
When asserting with `pm.response.to.not.be.*` before `pm.response.to.be.*`, the second assertion wrongly fails. However switching their order makes the tests correctly pass. Also, using `to.be.not` instead of `to.not.be` makes the tests work as well.

## Examples:

The following 2 snippets work with a `PASS` result:
 ```javascript
pm.test("Response should be okay", function () {
  pm.response.to.be.ok;
  pm.response.to.not.be.error;
});
```

```javascript
pm.test("Response should be okay", function () {
  pm.response.to.be.not.error;
  pm.response.to.be.ok;
});
```

However, the following assertion always fails:
```javascript
pm.test("Response should be okay", function () {
  pm.response.to.not.be.error;
  pm.response.to.be.ok;
});
```
The last script gives the following message:

> AssertionError:expected response to not have status reason 'OK'

The following screenshot shows a status code of `200` while the assertion fails.

![Postman Assertion Error](https://user-images.githubusercontent.com/530409/31220627-ad6ab774-a9c9-11e7-9083-293858b9312a.png)

## App Details:
```
Postman for Linux
Version 5.2.1
linux 4.10.0-35-generic / x64
```

## Issue Report:
 1. Did you encounter this recently, or has this bug always been there
  _I know it happens since 5.2.0. I hadn't tested it before this version._
 2. Expected behavior
  _Asserion should have passed_

**Comment by numaanashraf** [comment]
Thanks for reporting this @dannyhajj. We are looking into it.

**Comment by saiudayakella** [comment]
+1

**Comment by EricWafford** [comment]
Sorry, I didn't see this before even though I did try to search, but I have the same issue here: https://github.com/postmanlabs/postman-app-support/issues/3652


It will also fail other tests after the use of .not.be

**Comment by ray-qu** [comment]
+1

Another way to put this issue is as below based on my observation:

**SUCCESS:**
```
pm.test("response should be okay to process", function () { 
    pm.response.to.have.status(200);
    pm.response.to.not.be.error;  // status code changes post this line
    pm.response.to.have.status(201);
});
```

**FAILURE:**
```
pm.test("response should be okay to process", function () { 
    pm.response.to.have.status(200);
    pm.response.to.not.be.error;  // status code changes post this line
    pm.response.to.have.status(200);
});

```

**Comment by p00j4** [comment]
Is there any actionable in progress on this? I also ran into the same https://github.com/postmanlabs/postman-app-support/issues/4710

**Comment by desinole** [comment]
Any updates on this issue? It's almost a year since it was first reported and I'm running into the same issues today

**Comment by kamalaknn** [comment]
We've pushed a fix for this in the latest version Postman 6.7.1. Could you update and lets us know if you're still facing any issues?

**Comment by dannyhajj** [comment]
Works as expected now! Thanks!
